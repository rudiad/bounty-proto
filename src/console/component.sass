.console
  bottom: 0
  position: absolute
  border: 1px solid black
  padding: .7em
  left: 0
  right: 0
  margin-left: auto
  margin-right: auto
  height: 10em
  color: white
  background-color: #241d30
  opacity: 0.8
  overflow-y: auto
  font-family: "Lucida Console", Monaco, monospace
  z-index: -1
