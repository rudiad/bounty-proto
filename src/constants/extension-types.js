const { ExtensionAnchor, ExtensionPlatform } = window['extension-coordinator'];

const extensionTypes = {};
extensionTypes[ExtensionAnchor.Overlay] = "Overlay";
extensionTypes[ExtensionAnchor.Panel] = "Panel";
extensionTypes[ExtensionAnchor.Component] = "Component";
extensionTypes[ExtensionPlatform.Mobile] = "Mobile";

export const ExtensionAnchors = Object.freeze(extensionTypes);
export const DEFAULT_EXTENSION_TYPE = ExtensionAnchor.Overlay;
