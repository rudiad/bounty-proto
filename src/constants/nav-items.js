export const EXTENSION_VIEWS = 'Extension Views';
export const BROADCASTER_CONFIG = 'Broadcaster Config';
export const LIVE_CONFIG = 'Live Config';
export const CONFIGURATIONS = 'Configurations';
