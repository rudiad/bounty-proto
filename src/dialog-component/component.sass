$top-bar-dimension: 40px

.dialog__backdrop
  position: fixed
  top: 0
  bottom: 0
  left: 0
  right: 0
  background-color: #19171C
  opacity: 0.75
  padding: 50
  height: 100vh

.dialog
  background-color: white
  border-radius: 5
  width: 66%
  margin: 0 auto
  border: 1px solid #DAD8DE
  position: absolute
  top: 20%
  left: 20%

.container
  display: flex

.container__block
  display: inline-block

.container__left-block
  @extend .container__block
  float: left
  margin-left: 20px

.container__right-block
  @extend .container__block
  float: right

.top-bar
  @extend .container
  height: $top-bar-dimension
  justify-content: space-between

.top-bar-element
  width: auto
  text-align: center
  vertical-align: middle
  line-height: $top-bar-dimension
  font-weight: lighter

.top-bar-element-left
  @extend .container__left-block
  @extend .top-bar-element

.top-bar-element-right
  @extend .container__right-block
  @extend .top-bar-element

.dialog__divider-width
  width: 96%

@mixin bottom-bar-box($color, $bg-color)
  color: $color
  background-color: $bg-color
  border: 1px solid #7D55C7
  float: left
  text-align: center
  width: 80px
  height: 30px
  vertical-align: middle
  line-height: 30px
  cursor: pointer
  font-size: 16px
  margin-left: 20px
  margin-top: 10px
  margin-bottom: 30px

.title
  font-size: 13px
  font-weight: lighter
