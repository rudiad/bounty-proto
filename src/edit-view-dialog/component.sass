@import "../dialog-component/component.sass"

.edit-view__background
  @extend .dialog__backdrop

.edit-view__dialog-component
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__dialog-mobile
  @extend .dialog
  height: 275px
  width: 415px
  left: 35%

.edit-view__dialog-overlay
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__content
  @extend .container__left-block
  overflow-y: auto
  height: 50%
  width: 95%
  display: flex
  flex-direction: column

.edit-subcontainer__input
  margin-top: .5em
  margin-bottom: .5em

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.dialog__bottom-bar
  @extend .container__left-block

.bottom-bar__save
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
  margin-left: 10px
