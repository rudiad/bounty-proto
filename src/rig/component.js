import React, { Component } from 'react';
import { RigNav } from '../rig-nav';
import { ExtensionViewContainer } from '../extension-view-container';
import { ExtensionRigConsole } from '../console';
import { ExtensionViewDialog } from '../extension-view-dialog';
import { RigConfigurationsDialog } from '../rig-configurations-dialog';
import { EditViewDialog } from '../edit-view-dialog';
import { createExtensionObject } from '../util/extension';
import { createSignedToken } from '../util/token';
import { fetchManifest, fetchExtensionManifest } from '../util/api';
import { EXTENSION_VIEWS, BROADCASTER_CONFIG, LIVE_CONFIG, CONFIGURATIONS, MOBILE_VIEW } from '../constants/nav-items'
import { ViewerTypes } from '../constants/viewer-types';
import { OverlaySizes } from '../constants/overlay-sizes';
import { IdentityOptions } from '../constants/identity-options';
import { RIG_ROLE } from '../constants/rig';
const { ExtensionMode, ExtensionViewType } = window['extension-coordinator'];

export class Rig extends Component {
  constructor(props) {
    super(props);

    this.state = {
      clientId: process.env.EXT_CLIENT_ID,
      secret: process.env.EXT_SECRET,
      version: process.env.EXT_VERSION,
      channelId: process.env.EXT_CHANNEL_ID,
      userName: process.env.EXT_USER_NAME,
      mode: ExtensionMode.Viewer,
      extensionViews: [],
      manifest: {},
      showExtensionsView: false,
      showConfigurations: false,
      showEditView: false,
      idToEdit: 0,
      selectedView: EXTENSION_VIEWS,
      extension: {},
      userId: '',
      error: '',
    };
  }

  componentDidMount() {
    this._fetchInitialConfiguration();
  }

  componentWillMount() {
    this._initLocalStorage();
  }

  openConfigurationsHandler = () => {
    this.setState({
      showConfigurations: true,
      selectedView: CONFIGURATIONS
    });
  }
  closeConfigurationsHandler = () => {
    this.setState({
      showConfigurations: false,
    });
  }

  openEditViewHandler = (id) => {
    this.setState({
      showEditView: true,
      idToEdit: id,
    });
  }

  closeEditViewHandler = () => {
    this.setState({
      showEditView: false,
      idToEdit: '0',
    });
  }

  viewerHandler = () => {
    this.setState({
      mode: ExtensionMode.Viewer,
      selectedView: EXTENSION_VIEWS,
      extension: {},
    });
  }

  mobileHandler = () => {
    this.setState({
      mode: ExtensionViewType.Mobile,
      selectedView: MOBILE_VIEW,
      extension: createExtensionObject(
        this.state.manifest,
        0,
        ViewerTypes.Broadcaster,
        false,
        this.state.userName,
        this.state.channelId,
        this.state.secret),
    })
  }

  configHandler = () => {
    this.setState({
      mode: ExtensionMode.Config,
      selectedView: BROADCASTER_CONFIG,
      extension: createExtensionObject(
        this.state.manifest,
        0,
        ViewerTypes.Broadcaster,
        '',
        this.state.userName,
        this.state.channelId,
        this.state.secret),
    });
  }

  liveConfigHandler = () => {
    this.setState({
      mode: ExtensionMode.Dashboard,
      selectedView: LIVE_CONFIG,
      extension: createExtensionObject(
        this.state.manifest,
        0,
        ViewerTypes.Broadcaster,
        '',
        this.state.userName,
        this.state.channelId,
        this.state.secret),
    });
  }

  openExtensionViewHandler = () => {
    if (this.state.error === '') {
      this.setState({
        showExtensionsView: true,
      });
    }
  }

  closeExtensionViewDialog = () => {
    this.setState({
      showExtensionsView: false
    });
  }

  refreshConfigurationsHandler = () => {
    const token = createSignedToken(RIG_ROLE, '', this.state.userId, this.state.channelId, this.state.secret);
    fetchExtensionManifest('api.twitch.tv', this.state.clientId, this.state.version, token, this._onConfigurationSuccess, this._onConfigurationError);
  }

  _onConfigurationSuccess = (data) => {
    this.setState(data);
  }

  _onConfigurationError = (errMsg) => {
    this.setState({
      error: errMsg,
    });
  }

  createExtensionView = () => {
    const extensionViews = this._getExtensionViews();
    const linked = this.refs.extensionViewDialog.state.identityOption === IdentityOptions.Linked;
    extensionViews.push({
      id: (extensionViews.length + 1).toString(),
      type: this.refs.extensionViewDialog.state.extensionViewType,
      extension: createExtensionObject(
        this.state.manifest,
        (extensionViews.length + 1).toString(),
        this.refs.extensionViewDialog.state.viewerType,
        linked,
        this.state.userName,
        this.state.channelId,
        this.state.secret
      ),
      linked: linked,
      role: this.refs.extensionViewDialog.state.viewerType,
      x: this.refs.extensionViewDialog.state.x,
      y: this.refs.extensionViewDialog.state.y,
      overlaySize: (this.refs.extensionViewDialog.state.overlaySize === 'Custom' ? {width: this.refs.extensionViewDialog.state.width, height: this.refs.extensionViewDialog.state.height} : OverlaySizes[this.refs.extensionViewDialog.state.overlaySize]),
    });
    this._pushExtensionViews(extensionViews);
    this.closeExtensionViewDialog();
  }

  deleteExtensionView = (id) => {
    this._pushExtensionViews(this.state.extensionViews.filter(element => element.id !== id));
  }

  editComponentViewPosition = (position) => {
    const views = this._getExtensionViews();
    views.forEach(element => {
      if (element.id === this.state.idToEdit) {
        element.x = position.x;
        element.y = position.y;
      }
    });
    this._pushExtensionViews(views);
    this.closeEditViewHandler();
  }

  render() {
    return (
      <div>
        <RigNav
          ref="rigNav"
          selectedView={this.state.selectedView}
          viewerHandler={this.viewerHandler}
          configHandler={this.configHandler}
          liveConfigHandler={this.liveConfigHandler}
          openConfigurationsHandler={this.openConfigurationsHandler}
          error={this.state.error}/>
        <ExtensionViewContainer
          ref="extensionViewContainer"
          mode={this.state.mode}
          extensionViews={this.state.extensionViews}
          deleteExtensionViewHandler={this.deleteExtensionView}
          openExtensionViewHandler={this.openExtensionViewHandler}
          openEditViewHandler={this.openEditViewHandler}
          extension={this.state.extension} />
        {this.state.showExtensionsView &&
          <ExtensionViewDialog
            ref="extensionViewDialog"
            extensionViews={this.state.manifest.views}
            show={this.state.showExtensionsView}
            closeHandler={this.closeExtensionViewDialog}
            saveHandler={this.createExtensionView} />}
        {this.state.showEditView &&
          <EditViewDialog
            ref="editViewDialog"
            idToEdit={this.state.idToEdit}
            show={this.state.showEditView}
            views={this._getExtensionViews()}
            closeHandler={this.closeEditViewHandler}
            saveViewHandler={this.editComponentViewPosition}
          />}
        <RigConfigurationsDialog
          show={this.state.showConfigurations}
          config={this.state.manifest}
          closeConfigurationsHandler={this.closeConfigurationsHandler}
          refreshConfigurationsHandler={this.refreshConfigurationsHandler} />
        <ExtensionRigConsole />
      </div>
    );
  }

  _getExtensionViews() {
    const extensionViewsValue = localStorage.getItem("extensionViews");
    return extensionViewsValue ? JSON.parse(extensionViewsValue) : extensionViewsValue;
  }

  _pushExtensionViews(newViews) {
    localStorage.setItem("extensionViews", JSON.stringify(newViews));
    this.setState({
      extensionViews: newViews,
    });
  }

  _fetchInitialConfiguration() {
    fetchManifest("api.twitch.tv", this.state.clientId, this.state.userName, this.state.version, this.state.channelId, this.state.secret, this._onConfigurationSuccess, this._onConfigurationError);
  }

  _initLocalStorage() {
    const extensionViewsValue = localStorage.getItem("extensionViews");
    if (!extensionViewsValue) {
      localStorage.setItem("extensionViews", JSON.stringify([]));
      return;
    }
    this.setState({
      extensionViews: JSON.parse(extensionViewsValue)
    })
  }
}
