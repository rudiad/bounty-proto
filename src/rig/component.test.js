import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { createViewsForTest, ExtensionForTest } from '../tests/constants/extension';
import { mockFetchForManifest } from '../tests/mocks';
import { EXTENSION_VIEWS, BROADCASTER_CONFIG, LIVE_CONFIG } from '../constants/nav-items';
import { Rig } from './component';
import { ExtensionAnchors } from '../constants/extension-types';
import { ViewerTypes } from '../constants/viewer-types';
import { IdentityOptions } from '../constants/identity-options';
const { ExtensionMode, ExtensionAnchor } = window['extension-coordinator'];

describe('<Rig />', () => {
  const setupShallow = setupShallowTest(Rig, () => { });
  const setupViewsForTest = numViews => {
    const testViews = createViewsForTest(numViews, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    localStorage.setItem('extensionViews', JSON.stringify(testViews));
  };
  const setupComponentViewsForTest = numViews => {
    const testViews = createViewsForTest(numViews, ExtensionAnchors[ExtensionAnchor.Component], ViewerTypes.LoggedOut);
    localStorage.setItem('extensionViews', JSON.stringify(testViews));
  };

  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(mockFetchForManifest);
  })

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('gets extension views from local storage correctly when not in local storage', () => {
    const { wrapper } = setupShallow();
    expect(wrapper.instance()._getExtensionViews()).toEqual([]);
  });

  it('renders extension view correctly', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('ExtensionViewContainer').props().extensionViews).toHaveLength(1);
  });

  it('gets extension views from local storage correctly', () => {
    setupViewsForTest(1);
    const testViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    const { wrapper } = setupShallow();
    expect(wrapper.instance()._getExtensionViews()).toEqual(testViews);
  });

  it('deletes extension view correctly', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    expect(wrapper.find('ExtensionViewContainer').props().extensionViews).toHaveLength(1);

    wrapper.instance().deleteExtensionView('1');
    wrapper.update();
    expect(wrapper.find('ExtensionViewContainer').props().extensionViews).toHaveLength(0);
  });

  it('toggles state when edit dialog is opened/closed', () => {
    setupComponentViewsForTest(1);
    const { wrapper } = setupShallow();

    wrapper.instance().openEditViewHandler('1');
    expect(wrapper.instance().state.showEditView).toBe(true);
    expect(wrapper.instance().state.idToEdit).toBe('1');

    wrapper.instance().closeEditViewHandler('1');
    expect(wrapper.instance().state.showEditView).toBe(false);
    expect(wrapper.instance().state.idToEdit).toBe('0');
  });

  it('edit changes the view and sets them correctly', () => {
    setupComponentViewsForTest(1);
    const { wrapper } = setupShallow();

    wrapper.instance().openEditViewHandler('1');
    expect(wrapper.instance().state.showEditView).toBe(true);
    expect(wrapper.instance().state.idToEdit).toBe('1');

    wrapper.instance().editComponentViewPosition({ x: 25, y: 25 });

    const views = wrapper.instance()._getExtensionViews();
    const editedView = views.filter(element => element.id === '1');
    expect(editedView[0].x).toEqual(25);
    expect(editedView[0].y).toEqual(25);
    expect(wrapper.instance().state.showEditView).toBe(false);
    expect(wrapper.instance().state.idToEdit).toBe('0');
  });

  it('correctly toggles state when configuration dialog is opened/closed', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    wrapper.instance().openConfigurationsHandler();
    expect(wrapper.instance().state.showConfigurations).toBe(true);


    wrapper.instance().closeConfigurationsHandler();
    expect(wrapper.instance().state.showConfigurations).toBe(false);
  });

  it('correctly toggles state when create extension view is opened/closed', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    wrapper.setState({
      manifest: ExtensionForTest,
    })
    wrapper.instance().openExtensionViewHandler();
    expect(wrapper.instance().state.showExtensionsView).toBe(true);


    wrapper.instance().closeExtensionViewDialog();
    expect(wrapper.instance().state.showExtensionsView).toBe(false);
  });

  it('correctly sets state when viewHandler is invoked', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();

    wrapper.instance().viewerHandler();
    expect(wrapper.instance().state.mode).toBe(ExtensionMode.Viewer);
    expect(wrapper.instance().state.selectedView).toBe(EXTENSION_VIEWS);
    expect(wrapper.instance().state.extension).toEqual({});
  });


  it('correctly sets state when configHandler is invoked', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();

    wrapper.instance().configHandler();
    expect(wrapper.instance().state.mode).toBe(ExtensionMode.Config);
    expect(wrapper.instance().state.selectedView).toBe(BROADCASTER_CONFIG);
  });

  it('correctly sets state when liveConfigHandler is invoked', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    wrapper.instance().liveConfigHandler();
    expect(wrapper.instance().state.mode).toBe(ExtensionMode.Dashboard);
    expect(wrapper.instance().state.selectedView).toBe(LIVE_CONFIG);
  });

  it('gets the correct views when _getExtensionViews invoked', () => {
    const testViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    setupViewsForTest(1);
    const { wrapper } = setupShallow();

    expect(wrapper.instance().state.extensionViews).toEqual(testViews);
  });

  it('sets views in local storage correctly when _pushExtensionViews invoked', () => {
    const testViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    expect(wrapper.instance().state.extensionViews).toEqual(testViews);

    wrapper.instance().state.extensionViews.push(createViewsForTest(1,ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut));
    wrapper.instance()._pushExtensionViews(wrapper.instance().state.extensionViews);
  });

  it('sets state correctly when _onConfigurationSuccess invoked', () => {
    const { wrapper } = setupShallow();
    const testData = {
      test: 'test'
    };
    wrapper.instance()._onConfigurationSuccess(testData);
    expect(wrapper.instance().state.test).toBe(testData.test);
  });

  it('sets error state correctly when _onConfigurationError invoked', () => {
    const { wrapper } = setupShallow();
    const testError = 'error';
    wrapper.instance()._onConfigurationError(testError);
    expect(wrapper.instance().state.error).toBe(testError);
  });
});
