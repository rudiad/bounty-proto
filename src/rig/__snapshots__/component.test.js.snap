// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rig /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rig />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RigNav
          configHandler={[Function]}
          error=""
          liveConfigHandler={[Function]}
          openConfigurationsHandler={[Function]}
          selectedView="Extension Views"
          viewerHandler={[Function]}
/>,
        <ExtensionViewContainer
          deleteExtensionViewHandler={[Function]}
          extension={Object {}}
          extensionViews={Array []}
          mode="viewer"
          openEditViewHandler={[Function]}
          openExtensionViewHandler={[Function]}
/>,
        false,
        false,
        <RigConfigurationsDialog
          closeConfigurationsHandler={[Function]}
          config={Object {}}
          refreshConfigurationsHandler={[Function]}
          show={false}
/>,
        <ExtensionRigConsole />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "configHandler": [Function],
          "error": "",
          "liveConfigHandler": [Function],
          "openConfigurationsHandler": [Function],
          "selectedView": "Extension Views",
          "viewerHandler": [Function],
        },
        "ref": "rigNav",
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "deleteExtensionViewHandler": [Function],
          "extension": Object {},
          "extensionViews": Array [],
          "mode": "viewer",
          "openEditViewHandler": [Function],
          "openExtensionViewHandler": [Function],
        },
        "ref": "extensionViewContainer",
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeConfigurationsHandler": [Function],
          "config": Object {},
          "refreshConfigurationsHandler": [Function],
          "show": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RigNav
            configHandler={[Function]}
            error=""
            liveConfigHandler={[Function]}
            openConfigurationsHandler={[Function]}
            selectedView="Extension Views"
            viewerHandler={[Function]}
/>,
          <ExtensionViewContainer
            deleteExtensionViewHandler={[Function]}
            extension={Object {}}
            extensionViews={Array []}
            mode="viewer"
            openEditViewHandler={[Function]}
            openExtensionViewHandler={[Function]}
/>,
          false,
          false,
          <RigConfigurationsDialog
            closeConfigurationsHandler={[Function]}
            config={Object {}}
            refreshConfigurationsHandler={[Function]}
            show={false}
/>,
          <ExtensionRigConsole />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "configHandler": [Function],
            "error": "",
            "liveConfigHandler": [Function],
            "openConfigurationsHandler": [Function],
            "selectedView": "Extension Views",
            "viewerHandler": [Function],
          },
          "ref": "rigNav",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "deleteExtensionViewHandler": [Function],
            "extension": Object {},
            "extensionViews": Array [],
            "mode": "viewer",
            "openEditViewHandler": [Function],
            "openExtensionViewHandler": [Function],
          },
          "ref": "extensionViewContainer",
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeConfigurationsHandler": [Function],
            "config": Object {},
            "refreshConfigurationsHandler": [Function],
            "show": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Rig /> renders extension view correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rig />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RigNav
          configHandler={[Function]}
          error=""
          liveConfigHandler={[Function]}
          openConfigurationsHandler={[Function]}
          selectedView="Extension Views"
          viewerHandler={[Function]}
/>,
        <ExtensionViewContainer
          deleteExtensionViewHandler={[Function]}
          extension={Object {}}
          extensionViews={
                    Array [
                              Object {
                                "extension": Object {
                                  "authorName": "test",
                                  "channelId": "channelId",
                                  "description": "description",
                                  "iconUrl": "icon_url",
                                  "id": "id",
                                  "name": "name",
                                  "requestIdentity": false,
                                  "sku": "sku",
                                  "state": "state",
                                  "summary": "summary",
                                  "token": "token",
                                  "vendorCode": "vendorCode",
                                  "version": "0.1",
                                  "views": Object {
                                    "component": Object {
                                      "aspectHeight": 3000,
                                      "aspectWidth": 2500,
                                      "viewerUrl": "test",
                                      "zoom": false,
                                    },
                                    "config": Object {
                                      "viewerUrl": "test",
                                    },
                                    "liveConfig": Object {
                                      "viewerUrl": "test",
                                    },
                                    "panel": Object {
                                      "viewerUrl": "test",
                                    },
                                  },
                                  "whitelistedConfigUrls": Array [
                                    "foo",
                                  ],
                                  "whitelistedPanelUrls": Array [
                                    "bar",
                                  ],
                                },
                                "id": "1",
                                "linked": false,
                                "orientation": "Portrait",
                                "role": "Logged-Out Viewer",
                                "type": "Panel",
                                "x": 0,
                                "y": 0,
                              },
                            ]
          }
          mode="viewer"
          openEditViewHandler={[Function]}
          openExtensionViewHandler={[Function]}
/>,
        false,
        false,
        <RigConfigurationsDialog
          closeConfigurationsHandler={[Function]}
          config={Object {}}
          refreshConfigurationsHandler={[Function]}
          show={false}
/>,
        <ExtensionRigConsole />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "configHandler": [Function],
          "error": "",
          "liveConfigHandler": [Function],
          "openConfigurationsHandler": [Function],
          "selectedView": "Extension Views",
          "viewerHandler": [Function],
        },
        "ref": "rigNav",
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "deleteExtensionViewHandler": [Function],
          "extension": Object {},
          "extensionViews": Array [
            Object {
              "extension": Object {
                "authorName": "test",
                "channelId": "channelId",
                "description": "description",
                "iconUrl": "icon_url",
                "id": "id",
                "name": "name",
                "requestIdentity": false,
                "sku": "sku",
                "state": "state",
                "summary": "summary",
                "token": "token",
                "vendorCode": "vendorCode",
                "version": "0.1",
                "views": Object {
                  "component": Object {
                    "aspectHeight": 3000,
                    "aspectWidth": 2500,
                    "viewerUrl": "test",
                    "zoom": false,
                  },
                  "config": Object {
                    "viewerUrl": "test",
                  },
                  "liveConfig": Object {
                    "viewerUrl": "test",
                  },
                  "panel": Object {
                    "viewerUrl": "test",
                  },
                },
                "whitelistedConfigUrls": Array [
                  "foo",
                ],
                "whitelistedPanelUrls": Array [
                  "bar",
                ],
              },
              "id": "1",
              "linked": false,
              "orientation": "Portrait",
              "role": "Logged-Out Viewer",
              "type": "Panel",
              "x": 0,
              "y": 0,
            },
          ],
          "mode": "viewer",
          "openEditViewHandler": [Function],
          "openExtensionViewHandler": [Function],
        },
        "ref": "extensionViewContainer",
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeConfigurationsHandler": [Function],
          "config": Object {},
          "refreshConfigurationsHandler": [Function],
          "show": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RigNav
            configHandler={[Function]}
            error=""
            liveConfigHandler={[Function]}
            openConfigurationsHandler={[Function]}
            selectedView="Extension Views"
            viewerHandler={[Function]}
/>,
          <ExtensionViewContainer
            deleteExtensionViewHandler={[Function]}
            extension={Object {}}
            extensionViews={
                        Array [
                                    Object {
                                      "extension": Object {
                                        "authorName": "test",
                                        "channelId": "channelId",
                                        "description": "description",
                                        "iconUrl": "icon_url",
                                        "id": "id",
                                        "name": "name",
                                        "requestIdentity": false,
                                        "sku": "sku",
                                        "state": "state",
                                        "summary": "summary",
                                        "token": "token",
                                        "vendorCode": "vendorCode",
                                        "version": "0.1",
                                        "views": Object {
                                          "component": Object {
                                            "aspectHeight": 3000,
                                            "aspectWidth": 2500,
                                            "viewerUrl": "test",
                                            "zoom": false,
                                          },
                                          "config": Object {
                                            "viewerUrl": "test",
                                          },
                                          "liveConfig": Object {
                                            "viewerUrl": "test",
                                          },
                                          "panel": Object {
                                            "viewerUrl": "test",
                                          },
                                        },
                                        "whitelistedConfigUrls": Array [
                                          "foo",
                                        ],
                                        "whitelistedPanelUrls": Array [
                                          "bar",
                                        ],
                                      },
                                      "id": "1",
                                      "linked": false,
                                      "orientation": "Portrait",
                                      "role": "Logged-Out Viewer",
                                      "type": "Panel",
                                      "x": 0,
                                      "y": 0,
                                    },
                                  ]
            }
            mode="viewer"
            openEditViewHandler={[Function]}
            openExtensionViewHandler={[Function]}
/>,
          false,
          false,
          <RigConfigurationsDialog
            closeConfigurationsHandler={[Function]}
            config={Object {}}
            refreshConfigurationsHandler={[Function]}
            show={false}
/>,
          <ExtensionRigConsole />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "configHandler": [Function],
            "error": "",
            "liveConfigHandler": [Function],
            "openConfigurationsHandler": [Function],
            "selectedView": "Extension Views",
            "viewerHandler": [Function],
          },
          "ref": "rigNav",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "deleteExtensionViewHandler": [Function],
            "extension": Object {},
            "extensionViews": Array [
              Object {
                "extension": Object {
                  "authorName": "test",
                  "channelId": "channelId",
                  "description": "description",
                  "iconUrl": "icon_url",
                  "id": "id",
                  "name": "name",
                  "requestIdentity": false,
                  "sku": "sku",
                  "state": "state",
                  "summary": "summary",
                  "token": "token",
                  "vendorCode": "vendorCode",
                  "version": "0.1",
                  "views": Object {
                    "component": Object {
                      "aspectHeight": 3000,
                      "aspectWidth": 2500,
                      "viewerUrl": "test",
                      "zoom": false,
                    },
                    "config": Object {
                      "viewerUrl": "test",
                    },
                    "liveConfig": Object {
                      "viewerUrl": "test",
                    },
                    "panel": Object {
                      "viewerUrl": "test",
                    },
                  },
                  "whitelistedConfigUrls": Array [
                    "foo",
                  ],
                  "whitelistedPanelUrls": Array [
                    "bar",
                  ],
                },
                "id": "1",
                "linked": false,
                "orientation": "Portrait",
                "role": "Logged-Out Viewer",
                "type": "Panel",
                "x": 0,
                "y": 0,
              },
            ],
            "mode": "viewer",
            "openEditViewHandler": [Function],
            "openExtensionViewHandler": [Function],
          },
          "ref": "extensionViewContainer",
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeConfigurationsHandler": [Function],
            "config": Object {},
            "refreshConfigurationsHandler": [Function],
            "show": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
